feature-libs/cart/base/components/abstract-order-context/abstract-order-context-source.model.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+
+import { AbstractOrderType } from '@spartacus/cart/base/root';
+import { BehaviorSubject } from 'rxjs';
+import { AbstractOrderContext } from './abstract-order-context.model';
+
+/**
+ * Context source for AbstractOrderContext. Default values represent the context
+ * of an active cart
+ *
+ * AbstractOrderContext should be injected in child components.
+ */
+@Injectable()
+export class AbstractOrderContextSource implements AbstractOrderContext {
+  readonly type$ = new BehaviorSubject<AbstractOrderType>(
+    AbstractOrderType.CART
+  );
+  readonly id$ = new BehaviorSubject<string | undefined>(undefined);
```

Comment:
[MEDIUM]
In other Context Source objects (like `CartItemContextSource`) we're using `ReplaySubject(1)` instead of `BehaviorSubject` to avoid emitting the first empty/dummy value in case of early subscription. Do you think we could use `ReplaySubject(1)` here as well or there are reasons to use `BehaviorSubject`?

Reply:
@ChristophHi

Reply:
Previously I though I could control sending changes more fine granular for BehaviourSubject (but on a closer look: that was wrong), so will change it